lib/mongo/collection/view/readable.rb

```diff
@@ -574,7 +578,7 @@ def parallel_scan(cursor_count, options = {})
                          :coll_name => collection.name
                      }).execute(server)
             end
-            Cursor.new(self, result, server, session: session)
+            Cursor.new(self, result, server)
```

Comment:
If parallelCollectionScan was run with an explicit session the cursors all must use that same explicit session.

Reply:
Yep thanks.